docs/sql-ref-ansi-compliance.md

```diff
@@ -19,6 +19,87 @@ license: |
   limitations under the License.
 ---
 
+Spark SQL has two options to comply with the SQL standard: `spark.sql.ansi.enabled` and `spark.sql.storeAssignmentPolicy`.
+When `spark.sql.ansi.enabled` is set to `true` (`false` by default), Spark SQL follows the standard in basic behaviours (e.g., arithmetic operations, type conversion, and SQL parsing).
+Moreover, Spark SQL has an independent option to control implicit casting behaviours when inserting rows in a table.
+The casting behaviours are defined as store assignment rules in the standard.
+When `spark.sql.storeAssignmentPolicy` is set to `ANSI`, Spark SQL complies with the ANSI store assignment rules and this setting is a default value.
```

Comment:
Shall we summarize as a table like https://spark.apache.org/docs/3.0.0-preview2/configuration.html#application-properties ? That will be more easier to explain and to understand and to maintain(add new conf later or change the default value).

For example, the default value of `spark.sql.storeAssignmentPolicy` is not mentioned in this paragraph.

Reply:
Ah, I see. But, when I was writing this doc, I assumed the basic info about `spark.sql.storeAssignmentPolicy` would appear in the SQL Conf doc: https://github.com/apache/spark/pull/27459
Better to describe the two options  in a tabular format here, too?

Reply:
I am +1 with showing a table here, which can be more clear.

Reply:
ok